eng/pipelines/docs.yml

```diff
@@ -7,75 +7,103 @@ jobs:
     pool:
       vmImage: vs2017-win2016
     steps:
+      - pwsh: |
+          Invoke-WebRequest -MaximumRetryCount 10 -Uri "https://github.com/mono/api-doc-tools/releases/download/mdoc-5.7.4.9/mdoc-5.7.4.9.zip" `
+          -OutFile "mdoc.zip" | Wait-Process; Expand-Archive -Path "mdoc.zip" -DestinationPath "./mdoc/"
+        workingDirectory: $(Build.BinariesDirectory)
+        displayName: Download and Extract mdoc Zip
       - pwsh: |
           Invoke-WebRequest -MaximumRetryCount 10 -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
           -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "./docfx/"
         workingDirectory: $(Build.BinariesDirectory)
         displayName: Download and Extract DocFX
+      - pwsh: |
+          dotnet restore "$(Build.SourcesDirectory)/eng/docgeneration/assets/docgen.csproj" --packages docgenpackages
+          dotnet restore "$(Build.SourcesDirectory)/eng/service.proj" --packages docgenpackages
+        displayName: Restore Nuget Packages Needed
+      - pwsh: |
+          New-Item -Type dir dll-docs/my-api
+          New-Item -Type dir dll-docs/dependencies/my-api
+          New-Item -Type dir dll-xml-output
+          New-Item -Type dir dll-yaml-output
+          New-Item -Type dir docfx-output
+        displayName: Create Directories for Doc Generation
+        workingDirectory: $(Build.BinariesDirectory)
+      - task: DotNetCoreInstaller@2
+        displayName: "Use .NET Core sdk 3.0.100-preview5-011568"
+        inputs:
+          version: 3.0.100-preview5-011568
+      - pwsh: dotnet build eng/service.proj /p:IncludeTests=false /p:IncludeSamples=false /p:OutputPath=$(Build.ArtifactStagingDirectory)/dll-output /p:TargetFramework=netstandard2.0
+        displayName: Build Packages
+      - task: CopyFiles@2
+        displayName: 'Copy over Dlls'
+        inputs:
+          SourceFolder: '$(Build.ArtifactStagingDirectory)/dll-output'
+          TargetFolder: '$(Build.BinariesDirectory)/dll-docs/my-api'
+          contents: |
+            *.dll
+            *.xml 
+      - task: CopyFiles@2
```

Comment:
Is this copy necessary? Assuming you pass in the path I wouldn't expect it to be.

Reply:
Removed